src/composables/use-search.ts

```diff
@@ -0,0 +1,89 @@
+import { computed, ref, useRouter, watch } from "@nuxtjs/composition-api"
+
+import { useSearchStore } from "~/stores/search"
+import { useMediaStore } from "~/stores/media"
+import { useI18nResultsCount } from "~/composables/use-i18n-utilities"
+import { useMatchSearchRoutes } from "~/composables/use-match-routes"
+import { ALL_MEDIA } from "~/constants/media"
+
+export const useSearch = () => {
+  const mediaStore = useMediaStore()
+  const searchStore = useSearchStore()
+  const router = useRouter()
+
+  const { matches: isSearchRoute } = useMatchSearchRoutes()
+
+  const storeSearchTerm = computed(() => searchStore.searchTerm)
+  /**
+   * To update the local search term when the route changes, when, for example,
+   * the user clicks the back button, we need to watch the store search term.
+   */
+  watch(storeSearchTerm, (newSearchTerm) => {
+    searchTerm.value = newSearchTerm
+  })
+
+  const localSearchTerm = ref(storeSearchTerm.value)
+
+  /**
+   * Search term has a getter and setter to be used as a v-model.
+   * To prevent sending unnecessary requests, we also keep track of whether
+   * the search term was changed.
+   */
+  const searchTerm = computed({
+    get: () => localSearchTerm.value,
+    set: (value: string) => {
+      localSearchTerm.value = value
+    },
+  })
```

Comment:
I was confused by this computed because I was surprised that a regular `ref` didn't behave the same way this particular `computed` does. To try to understand it I was playing with this code locally and was able to rename `localSearchTerm` to `searchTerm` and update all usages of `localSearchTerm` to `searchTerm` in this function and things continued to work fine. I couldn't reproduce the bug and when I ran the e2e tests locally it appeared to work as well :open_mouth: I haven't worked with Vue closely in a while though, so I might be missing something. Is there an edge case this covers that isn't in the tests? I didn't try much other than regular searches and the bug in the issue so I could be missing the case where this computed is important.

Reply:
Vue 2 recommended to use the writable getter for `v-model`. In Vue 3, the `v-model` implementation has slightly changed, and the docs describe a different way of adding vmodel to component (https://vuejs.org/guide/components/events.html#usage-with-v-model). I think it's best to refactoring this after we migrate to Nuxt 3.

Reply:
> Do you have any resources where I can read more about "Vue 2 recommended to use the writable getter for v-model"?
I remember reading it in 2020, but I couldn't find it on the docs site :( I tried removing the computed getter `searchTerm`, but then searching from the single result page stopped working. 
I think that the reason is that when `v-model` changes, under the hood  it uses `searchTerm = newValue`. With a writable computed, it updates its value. However, with a `ref`, this doesn't work, you would need to use `searchTerm.value = newValue`.